### PR TITLE
feat: Chat in Dashboard | Connected Users Display

### LINKED ISSUE #602: [FEATURE] Chat In Dashboard | Connected Users Display
### Detailed description

https://dev.thecyberhub.org/#/dashboard/chat

Login --> Dashboard --> Chat

Only show users that are in connection with you in chat search and display
When user is a connection : If user accepted different user's request.

### Context

Users could chat only with users that are connected one with another

### Possible implementation

Need to show under Users only users that you connected with

image

### Additional information

The "connected users" state can be found in src/features/connections, we use Redux.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by sumanjeet0012**
## Fixes Issue

closes #602 

## Changes proposed

created two new arrays for online Connections and offline Connections and used these arrays in place of filtered online connections and filtered online connections.

## Code of Conduct

- [x] By submitting this pull request, I confirm I've read and complied with the [CoC](https://github.com/thecyberworld/Support/blob/main/CODE_OF_CONDUCT.md) 🖖

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

---

You can also join our [Discord](https://discord.gg/QHBPq6xP5p) community.
Feel free to check out other cool repositories of the [Thecyberworld](https://github.com/thecyberworld).
Join the Thecyberworld GitHub Organisation by raising an [issue](https://github.com/thecyberworld/Support/issues/new?assignees=&labels=invite+me+to+the+organisation&template=invitation.yml&title=Please+invite+me+to+the+GitHub+Community+Organization) (you will be sent an invitation).

### CODE REVIEW

File: src/components/Chat/Sidebar/Sidebar.jsx [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
 import SearchInputBox from "../../Common/SearchInputBox";
 import { FaAngleDown, FaAngleUp } from "react-icons/fa";
+import { useDispatch, useSelector } from "react-redux";
+import { getConnections } from "../../../features/connections/connectionSlice";
 
 const Sidebar = ({ hideSidebar, onlinePeople, offlinePeople, selectedUserId, setSelectedUserId }) => {
+    const dispatch = useDispatch();
+    const { user } = useSelector((state) => state.auth);
+    const connections = useSelector((state) => state.connectionData);
+    const allConnections = connections?.connections?.connections?.map((connection) => connection.user);
```
Comment (ArkadiK94) [inline src/components/Chat/Sidebar/Sidebar.jsx:13]:
```suggestion
    const {connections, ...}= useSelector((state) => state.connectionData);
    const allConnections = connections?.connections?.map((connection) => connection.user);
```
We have a `connections` model in back-end in this we have a `connections` array which already confusing . Using it three times make it even more confusing. You can use destructuring here.

Comment (ArkadiK94) [inline src/components/Chat/Sidebar/Sidebar.jsx:13]: You forgot to check if `isAccepted` is true so when user only sends connection request, the "connection user" appear in the chat before the other side accepting or rejecting the request.

**Review by ArkadiK94 — CHANGES_REQUESTED** [review]
@sumanjeet0012 Looks great, I have two comments about it , the second is crucial.

File: src/components/Chat/Sidebar/Sidebar.jsx [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
 import Contact from "../Contact";
 import SearchInputBox from "../../Common/SearchInputBox";
 import { FaAngleDown, FaAngleUp } from "react-icons/fa";
+import { useDispatch, useSelector } from "react-redux";
+import { getConnections } from "../../../features/connections/connectionSlice";
 
 const Sidebar = ({ hideSidebar, onlinePeople, offlinePeople, selectedUserId, setSelectedUserId }) => {
+    const dispatch = useDispatch();
+    const { user } = useSelector((state) => state.auth);
+    const {connections, ...}= useSelector((state) => state.connectionData);
```
Comment (ArkadiK94) [inline src/components/Chat/Sidebar/Sidebar.jsx:12]:
```suggestion
    const {connections}= useSelector((state) => state.connectionData);
```
@sumanjeet0012 Sorry, I confused you, I meant by three dots if you need to destructure other variables so write them. you don't need those dots.

**Review by ArkadiK94 — CHANGES_REQUESTED** [review]

File: src/components/Chat/Sidebar/Sidebar.jsx (hunk 2 above)
Comment (sumanjeet0012) [reply src/components/Chat/Sidebar/Sidebar.jsx:12]: ok, currently i don't need those.

Comment (ArkadiK94) [reply src/components/Chat/Sidebar/Sidebar.jsx:12]: ok, so you can remove the dots :)

**Pull request merged**